### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by AnsarIbrahim**
# This PR Contains:

# Project requirements
- Add validations to User and Post what they require✅
- Write unit test for all models✅

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User'
+  validates :title, presence: true, length: { minimum: 5 }
```
Comment (Lucas-Masaba) [inline app/models/post.rb]: - [ ] According to the project requirements, the length should be a maximum of 250 characters
image

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    it 'is invalid with a negative post_counter' do
+      user = User.new(name: 'Ansar', post_counter: -1)
+      expect(user).to_not be_valid
+    end
+
+    it 'is valid with a post_counter of 0' do
+      user = User.new(name: 'Ansar', post_counter: 0)
+      expect(user).to be_valid
+    end
```
Comment (Lucas-Masaba) [inline spec/models/user_spec.rb:49]: - [ ] As mentioned in the project requirements, any custom methods in the models should have a corresponding test. Please test the `recent_posts` model in your User model.

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  it 'is invalid with a negative like_counter' do
+    post = Post.new(title: 'Test Title', author: user, like_counter: -1)
+    expect(post).to_not be_valid
+  end
+
+  it 'increments the author\'s post_counter after save' do
+    Post.create(title: 'Test Title', author: user)
+    expect(user.reload.post_counter).to eq(1)
```
Comment (Lucas-Masaba) [inline spec/models/post_spec.rb:38]: - [ ] Please add tests for the `recent_comments` and `update_posts_counter` methods found in your Post model.

File: spec/models/like_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      like = Like.new(user_id: user.id, post_id: post.id)
+      like.post_id = 'string'
+      expect(like).to_not be_valid
+    end
+
+    it 'should have an integer user_id' do
+      like = Like.new(user_id: user.id, post_id: post.id)
+      like.user_id = 'string'
+      expect(like).to_not be_valid
+    end
```
Comment (Lucas-Masaba) [inline spec/models/like_spec.rb:18]: - [ ] Please add tests for your `update_likes` method in your Like model

File: spec/models/comment_spec.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      comment = Comment.new(user_id: user.id, post_id: post.id, text: 'this is comment')
+      comment.post_id = nil
+      expect(comment).to_not be_valid
+    end
+
+    it 'user id must be integer' do
+      comment = Comment.new(user_id: user.id, post_id: post.id, text: 'this is comment')
+      comment.user_id = nil
+      expect(comment).to_not be_valid
+    end
```
Comment (Lucas-Masaba) [inline spec/models/comment_spec.rb:21]: - [ ] Please add a test for your `update_comments_counter` method in your Comment model.

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @AnsarIbrahim  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- All written tests are passing :heavy_check_mark: 
- Successfully added validations to your models :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Owusu-Desmond — APPROVED** [review]
Hi @AnsarIbrahim , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**